lab-anthony/model/user.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const mongoose = require('mongoose');
+const bcrypt = require('bcrypt');
+const httpErrors = require('http-errors');
+const crypto = require('crypto');
+const jwt = require('jsonwebtoken');
+const debug = require('debug')('coffee:user');
+
+const UserSchema = mongoose.Schema({
+  username: {type: String, required: true, unique: true},
+  password: {type: String, required: true},
+  findHash: {type: String, unique: true}
+});
+
+UserSchema.methods.generateHash = function(password){
+  debug('generateHash');
+  return new Promise((resolve, reject)=>{
+    bcrypt.hash(password, 8, (err, hash)=>{
+      this.password = hash;
+      resolve(this);
```

Comment:
I love that you're resolving with this! Awesome!

Shouldn't you do something to deal with the error if it comes up though?